src/Ocelot/RateLimiting/Middleware/RateLimitingMiddleware.cs

```diff
@@ -10,15 +12,18 @@ public class RateLimitingMiddleware : OcelotMiddleware
     {
         private readonly RequestDelegate _next;
         private readonly IRateLimiting _limiter;
+        private readonly IServiceProvider _container;
 
         public RateLimitingMiddleware(
             RequestDelegate next,
             IOcelotLoggerFactory factory,
-            IRateLimiting limiter)
+            IRateLimiting limiter,
+            IServiceProvider container)
```

Comment:
Injecting the service provider instead of directly injecting the `IHttpContextAccessor` object can be considered a significant design flaw, in my opinion.

Reply:
and it is a bad practice to embeded service provider

Reply:
Fixed in commit https://github.com/ThreeMammals/Ocelot/pull/1307/commits/42e7703385a74cd20ee30e2861829ea1e4de3750 ✔️